platform/ext/target/nordic_nrf/common/spm_hal.c

```diff
@@ -321,3 +321,68 @@ enum tfm_plat_err_t tfm_spm_hal_nvic_interrupt_enable(void)
 {
     return nvic_interrupt_enable();
 }
+
+bool tfm_spm_hal_has_access_to_region(const void *p, size_t s,
+                                              int flags)
+{
+    cmse_address_info_t tt_base = cmse_TT((void *)p);
+    cmse_address_info_t tt_last = cmse_TT((void *)((uint32_t)p + s - 1));
+
+    uint32_t base_spu_id = tt_base.flags.idau_region;
+    uint32_t last_spu_id = tt_last.flags.idau_region;
+
+    size_t size;
+    uint32_t p_start = (uint32_t)p;
+    int i;
+
+    if ((base_spu_id >= spu_regions_flash_get_start_id()) &&
+        (last_spu_id <= spu_regions_flash_get_last_id())) {
+
+        size = spu_regions_flash_get_last_address_in_region(base_spu_id) + 1 - p_start;
+
+        if (cmse_check_address_range((void *)p_start, size, flags) == 0) {
```

Comment:
Wouldn't it be sufficient to check only one byte in each region (permissions for other bytes within the same region are the same, right)? This would simplify the implementation quite a bit and also `cmse_check_address_range()` would execute faster then (since regions are aligned at 32-byte boundary, when the whole checked address range fits into one 32-byte block, the function executes only one TT instruction).

Reply:
No, @anangl we really need to check all region(s) because we don't know if the access-check is failing due to crossing IDAU boundaries, or due to violating MPU or SAU configuration. 

By checking within each of the SPU regions we know that any checking failures should be reported to the caller as legitimate.

Reply:
Right, I haven't thought about MPU in this context.